### ISSUE TITLE
UI for entry creation, editing and registration with seamless GitHub integration

### ISSUE THREAD

**Issue by joncison** [bio-tools/content#6]
One of many issues around [GitHub-based content management ](https://github.com/bio-tools/content/issues/1 )for _bio.tools_.

**Comment by joncison** [comment]
@hansioan - @hmenager and I discussed this would be a good place to start on the GitHub integration work, and nice collab. between DK & FR - starting by discussing the technicalities / requirements around this.  Hans, I'm not proposing you work on this yet, just that we start thinking about it more, with Herve.

General issues (not specifically required for GitHub integration) for edamToolAnnotator _e.g._  improving the edam browsing widget  go [here](https://github.com/bio-tools/edamToolAnnotator/issues)   .   

Use this thread specifically for issues arising from GitHub integration _per se_  In my mind, in very broad strokes, something like:

* Extension to incorporate all other data entry fields currently implemented in _bio.tools_ registration interface, including "JSON view"?
* Validation mechanism - initially using existing _bio.tools_ mechanism ?
* A "Load" button to expose a list of tool IDs & names (retrieved from GitHub), allow for search / auto-completion by ID & name, download the XML from GitHub, pre-populate the widgets with values
* A "Save" button to issue a pull request / push directly to GitHub ? (depending on privileges?)

cc @piotrgithub1

**Comment by piotrgithub1** [comment]
Just to add - please do not commit to any one format, the discussion hasn't been settled yet between prettified JSON and YAML.

**Comment by joncison** [comment]
Definitely.  My view to begin (I mean while it's all still a "sandbox") is to go with XML (for ease of validation).  Or JSON if that's more practical for whatever reason.  That entire discussion can be taken in https://github.com/bio-tools/content/issues/7.

**Comment by hmenager** [comment]
Thanks @joncison and @piotrgithub1.
Here's my two cents:
- format: start with JSON, as it is probably what edamToolAnnotator currently produces, and what bio.tools produces. XML, from experience, is a real pain to produce directly from as Js interface. I would rather, later, use a service to convert json to XML or YAML, and the other way around.
- validation: use a validation service.
- i wouldn't put much effort at first into loading a "list" of tools, rather I would expect that this component would be able to read and write to a github repo directly using the API

So, my "priority wishlist" for this is: 
- a component that can read, edit, and write complete bio.tools-compatible tool descriptions. 
- read and write operations would be primarily targetting github repositories using the API.
- validation and format conversion would use new or existing validation services from bio.tools.
- I would reuse the JSON format used for edam-browser and bio.tools to display EDAM.
I know it probably doesn't look very ambitious, but I think having a robust and user-friendly UI that provides all of this will already be very nice.

**Comment by joncison** [comment]
Thanks @hmenager, agree with all but a couple points in reply, cc @piotrgithub1 @bgruening @hansioan FTI

* format: JSON could work let's discuss also the [transforms](https://github.com/bio-tools/content/issues/7) and the [validation](https://github.com/bio-tools/content/issues/5)
* it would need to load a list of tools (_i.e._ whats in GitHub) in order to provide an editing function (over existing tools), wouldn't it?

**Comment by hmenager** [comment]
Hello @joncison @piotrgithub1 @hansioan, here's an update from the "FR" team. @ValentinMarcon has been working to create a proof-of-concept interface that instead of updating a DB would save created/modified tool descriptions as pull requests to a github repository. Of course, this is very much a work in progress, but the technical bases are functional, and we would like a lot to discuss it with you some time in the near future. The code is there: 
https://github.com/ValentinMarcon/biotools_to_github/
Feedback is welcome!

**Comment by joncison** [comment]
Nice! Looking forward to learning more (or even demo?) next week.  Sooner rather than later it would be good to move the project under https://github.com/bio-tools/ if you're OK with that.

An obvious thing would be to incorporate the best ideas or even the code if possible/sensible from [edamToolAnnotator](https://github.com/bio-tools/edamToolAnnotator)

**Comment by mihai-sysbio** [comment]
The gateway for individual tool edits is now the RSEc Atlas - each tool page has a button to "View on GitHub".

In terms of creation, I would argue that is best being left for individual source registries. Otherwise this can be seen as competing functionality, which is a duplication of effort. In terms of registration into the RSEc, that is anyway achieved through registries so again would not need any further development.

With this, I am proceeding to close this issue and if need be to open a new discussion.
